IntelPython/dpbench
# Fix multiprocessing

Fixes:
- Cross Framework import withing one process
- GPU runs?
- Windows run (except validation)
- cross process sycl initialization/execution

New features:
- `--no-validation` argument now actually works
- `--skip-expected-failure` argument was added

Bugs:
- for some environment validation does not work because it is done through 'fork'. It may not work from 'spawn' process

Infrastructure changes:
- Add new `BenchmarkRunner` that spawn process for each Framework to prevent cross import for frameworks run.
- Old `BecnhmarkRunner` still exists, because of the way how `Benchmark` validates the data. Refactoring of validation required. (will be done if following PRs)
- Each process sets up separate python runtime, which means that global variable not accessible between processes
- New process sets up GLOBAL config with only required frameworks/benchmark/implementations (should be refactored, this is the least painful way to update mp workflow)

This changes unblocks following features:
- Run benchmarks in single process (e.g. when debugging single benchmark implementation execution) with few lines of code change.
- Validation refactor (currently we run reference implementation for every implementation we want to validate)
- Many more

Checks:
- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to an issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?
